packages/@ember/-internals/runtime/lib/mixins/array.ts

```diff
@@ -22,17 +22,26 @@ import { ENV } from '@ember/-internals/environment';
 import Observable from '../mixins/observable';
 import MutableEnumerable from './mutable_enumerable';
 import { typeOf } from '../type-of';
+import type { AnyFn } from '@ember/-internals/utils/types';
+import type { ComputedPropertyCallback } from '@ember/-internals/metal/lib/computed';
 
-const EMPTY_ARRAY = Object.freeze([]);
+type Value<T, K extends string> = K extends keyof T ? T[K] : unknown;
 
-const identityFunction = (item) => item;
+// SAFETY: Nothing unsafe here, just TS can't infer the specificity we can.
+const EMPTY_ARRAY = Object.freeze([]) as readonly [];
```

Comment:
This is a great place to use `as const`, which doesn't require the `SAFETY` annotation (`as const` is never an unsafe cast) and gives *exactly* the desired type ([playground][p]).

```suggestion
const EMPTY_ARRAY = Object.freeze([] as const);
```

[p]: https://www.typescriptlang.org/play?#code/MYewdgzgLgBAogWQAoBUCaB9AggJR1tGAXhgHkAjAKwFNgoA6AMwCdrqAvagCgG0BdGAEMIMUJCgBKANxA

Reply:
Great, thanks!